### PR TITLE
[Enhancement] Support some date function transform for trino parser

### PR DESCRIPTION
**Pull request by Youngwb**
## What type of PR is this：
- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

#14825

## Problem Summary(Required) ：

support transform trino function `to_unixtime`, `date_parse`,`day_of_week`,`dow`,`day_of_month`,`day_of_year`,`doy` to starrocks function

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [x] 3.0
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 18 / 20 (90.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/connector/parser/trino/AstBuilder.java|2|4|50.00%|[784, 785]|
|:large_blue_circle:|com/starrocks/connector/parser/trino/Trino2SRFunctionCallTransformer.java|16|16|100.00%|[]|

File: fe/fe-core/src/main/java/com/starrocks/connector/parser/trino/Trino2SRFunctionCallTransformer.java [hunk 1]
```diff
@@ -122,6 +123,36 @@ private static void registerArrayFunctionTransformer() {
...
+        // day_of_month -> dayofmonth
+        registerFunctionTransformer("day_of_month", 1, "dayofmonth",
+                ImmutableList.of(Expr.class));
+
+        // day_of_year -> dayofyear
+        registerFunctionTransformer("day_of_year", 1, "dayofyear",
+                ImmutableList.of(Expr.class));
+
+        // doy -> dayofyear
+        registerFunctionTransformer("doy", 1, "dayofyear",
```
Comment (stephen-shelby) [inline fe/fe-core/src/main/java/com/starrocks/connector/parser/trino/Trino2SRFunctionCallTransformer.java:152]: doy means day of year?

Comment (Youngwb) [reply fe/fe-core/src/main/java/com/starrocks/connector/parser/trino/Trino2SRFunctionCallTransformer.java:152]: yes

**Review by stephen-shelby — APPROVED** [review]
LGTM

**Review by ABingHuang — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.0

**Pull request merged**